### PR TITLE
Improvement/rsmr210 refactor rsm predict

### PR DESCRIPTION
**Pull request by aloukina**
- Took the preprocessing code out of rsmpredict into a separate function
- Exposed it via API
- Used `process predictions` instead of duplicating code in rsmpredict
- Minor edits to the documentation
- Changed the header in RSMSummarize to use the common sorting script

### CODE REVIEW

File: rsmtool/__init__.py [hunk 1]
```diff
@@ -39,7 +39,8 @@
                          preprocess_feature,
                          remove_outliers,
                          transform_feature,
-                         trim)
+                         trim,
+                         preprocess_new_feature_file)
```
Comment (desilinguist) [inline rsmtool/__init__.py]: alphabetically sort?

File: rsmtool/preprocess.py [hunk 2]
```diff
@@ -746,3 +743,151 @@ def preprocess_train_and_test_features(df_train, df_test, feature_specs):
...
             df_feature_info)
+
+
+def preprocess_new_feature_file(df_input,
+                                df_feature_info):
+
+    """
+    Process a data frame with feature values by applying
+    :ref:`preprocessing parameters <preprocessing_parameters>`
+    stored in `df_feature_info`
```
Comment (desilinguist) [inline rsmtool/preprocess.py]: period at the end.

File: rsmtool/preprocess.py [hunk 3]
```diff
@@ -746,3 +743,151 @@ def preprocess_train_and_test_features(df_train, df_test, feature_specs):
     return (df_train_preprocessed,
             df_test_preprocessed,
             df_feature_info)
+
+
+def preprocess_new_feature_file(df_input,
```
Comment (desilinguist) [inline rsmtool/preprocess.py]: It's weird to call it `preprocess_new_file` when the input is a data frame?

**Comment by aloukina** [comment]
I am done with the changes to this PR

**Pull request merged**